src/Microsoft.Extensions.Localization/ResourceManagerStringLocalizer.cs

```diff
@@ -197,17 +204,21 @@ private IEnumerable<string> GetResourceNamesFromCultureHierarchy(CultureInfo sta
             var currentCulture = startingCulture;
             var resourceNames = new HashSet<string>();
 
+            var hasAnyCultures = false;
```

Comment:
Cant this also be derived by the size of `resourceNames`? Aka if it has at least 1 element